gradient_utils/metrics.py

```diff
@@ -53,6 +54,49 @@ def get_workload_id():
         return os.getenv(WORKLOAD_ID_ENV)
     return _get_experiment_id()
 
+def add_metrics(metrics, workload_id=None, registry=REGISTRY, push_gateway=None, timeout=30):
```

Comment:
I would drop `workload_id`, `registry`, and `push_gateway` from this interface. If a user wants that level of control they should use the underlying prom implementation.